### PR TITLE
[C#] feat: `AssistantsPlanner` support for Azure Assistants API

### PR DESCRIPTION
**Pull request by singhk97**
## Linked issues

closes: #1499 (issue number)

## Details

Introduces Azure OpenAI support to the `AssistantPlanner`

#### Change details

* Replaced internal client with `AssistantsClient` from `Microsoft.AI.OpenAI.Assistants` library
* Removed `OpenAIClient.Assistant.cs`, `OpenAIClient.Thread.cs` and all internal models used to facilitate call to Assistants API. Marked public models as `Obsolete`.

## Attestation Checklist

- [x] My code follows the style guidelines of this project

- I have checked for/fixed spelling, linting, and other errors
- I have commented my code for clarity
- I have made corresponding changes to the documentation (updating the doc strings in the code is sufficient)
- My changes generate no new warnings
- I have added tests that validates my changes, and provides sufficient test coverage. I have tested with:
  - Local testing
  - E2E testing in Teams
- New and existing unit tests pass locally with my changes

### Additional information

> Feel free to add other relevant information below

### CODE REVIEW

**Review by aacebo — APPROVED** [review]

File: dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI.Tests/TestUtils/TestAssistantsOpenAIClient.cs [hunk 1]
```diff
@@ -142,32 +141,63 @@ public override Task<Run> CreateRunAsync(string threadId, RunCreateParams runCre
...
+                run.Tools
+            );
+
+            Runs[threadId][i] = runWithUpdatedStatus;
+
+            return Task.FromResult(Response.FromValue(runWithUpdatedStatus, Mock.Of<Response>()));
         }
 
-        public override Task<Run?> RetrieveLastRunAsync(string threadId, CancellationToken cancellationToken)
+        public override async Task<Response<PageableList<ThreadRun>>> GetRunsAsync(string threadId, int? limit = null, ListSortOrder? order = null, string after = null, string before = null, CancellationToken cancellationToken = default)
```
Comment (corinagum) [inline dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI.Tests/TestUtils/TestAssistantsOpenAIClient.cs:177]: Could you handle these warnings listed below?

File: dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI.Tests/TestUtils/TestAssistantsOpenAIClient.cs [hunk 2]
```diff
@@ -34,105 +34,104 @@ public void Reset()
...
                 Messages.Add(threadId, new() { newMessage });
-            }
+            };
 
-            return Task.FromResult(newMessage);
+            return Task.FromResult(Response.FromValue(newMessage, Mock.Of<Response>()));
         }
 
-        public override async IAsyncEnumerable<Message> ListNewMessagesAsync(string threadId, string? lastMessageId, [EnumeratorCancellation] CancellationToken cancellationToken)
+        public override async Task<Response<PageableList<ThreadMessage>>> GetMessagesAsync(string threadId, int? limit = null, ListSortOrder? order = null, string after = null, string before = null, CancellationToken cancellationToken = default)
```
Comment (corinagum) [inline dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI.Tests/TestUtils/TestAssistantsOpenAIClient.cs:97]: Could you handle these warnings listed below?

File: dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI.Tests/TestUtils/TestAssistantsOpenAIClient.cs (hunk 1 above)
Comment (singhk97) [reply dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI.Tests/TestUtils/TestAssistantsOpenAIClient.cs]: fixed

File: dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI.Tests/TestUtils/TestAssistantsOpenAIClient.cs (hunk 2 above)
Comment (singhk97) [reply dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI.Tests/TestUtils/TestAssistantsOpenAIClient.cs]: fixed

File: dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI.Tests/TestUtils/TestAssistantsOpenAIClient.cs [hunk 3]
```diff
@@ -34,105 +34,104 @@ public void Reset()
...
+                messages.Add(newMessage);
+            }
+
+            Messages.Add(newThread.Id, messages);
+            Runs.Add(newThread.Id, new List<ThreadRun>());
+            return Task.FromResult(Response.FromValue(newThread, Mock.Of<Response>()));
         }
 
-        public override Task<Message> CreateMessageAsync(string threadId, MessageCreateParams messageCreateParams, CancellationToken cancellationToken)
+        public override Task<Response<ThreadMessage>> CreateMessageAsync(string threadId, MessageRole role, string content, IEnumerable<string> fileIds = null, IDictionary<string, string> metadata = null, CancellationToken cancellationToken = default)
```
[the remark this answers is not in the document]
Comment (singhk97) [reply dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI.Tests/TestUtils/TestAssistantsOpenAIClient.cs]: fixed

**Review by corinagum — APPROVED** [review]

**Pull request merged**